aws/cloudformation/domain_redirect.yml.erb

```diff
@@ -0,0 +1,146 @@
+<%
+  require 'aws-sdk-route53'
+
+  # Types of redirected domains
+  # - domain.com
+  # - sub.domain.com NOT SUPPORTED
+  # - sub.comain.com/with/path NOT SUPPORTED
+
+  # Related Resources
+  # - Domain Registrations: Not manageable by Cloudformation
+
+  raise "'domains' parameter must be provided" unless defined?(domains)
+  raise "'redirect_to' parameter must be provided" unless defined?(redirect_to)
+
+  domains = domains.split(',')
+    .map { |d| d.downcase }
+    .each { |d|
+      # lazily check that all domains are of the form something.tld
+      # - subdomains not currently supported
+      # - the longest TLD as of 7/15/2022 is 24 characters :O
+      if !d.match(/^[a-z0-9\-]+\.[a-z]{2,24}$/)
+        raise "'#{d}' is not a supported domain"
+      end
+    }
+
+    def camelize_domain(domain)
+      domain.split(/\./).map(&:capitalize).join("Dot")
```

Comment:
hah, `join("Dot")` is cute. might be worth adding a comment here just to make it explicit why we're doing this, though

Reply:
Added a comment explaining why we need CodeDotOrg instead of code.org.